src/xo-app/backup/restore/index.js

```diff
@@ -98,16 +103,90 @@ const doImport = ({ backup, sr, start }) => {
   }
 }
 
+class _Collapsible extends Component {
+  render () {
+    const {
+      collapsible,
+      children,
+      ...props
+    } = this.props
+
+    return collapsible
+      ? <Collapse {...props}> {children} </Collapse>
+      : <div>
+        <span> {props.buttonText} </span>
+        <br />
+        {children}
+      </div>
+  }
+}
+
 class _ModalBody extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      mapVdisSrs: {}
+    }
+  }
+
   get value () {
     return this.state
   }
 
+  _getSrPredicate = createSelector(
+    () => this.state.sr,
+    () => this.state.mapVdisSrs,
+    (defaultSr, mapVdisSrs) => sr => {
+      let predicate = isSrWritable(sr) && defaultSr.$pool === sr.$pool
```

Comment:
This should not be called `predicate`. A predicate is a function. Here, the predicate is `sr => { ... }`.